src/components/utils/src/timer.cc

```diff
@@ -135,37 +59,75 @@ timer::Timer::Timer(const std::string& name, const TimerTask* task_for_tracking)
 timer::Timer::~Timer() {
   LOG4CXX_AUTO_TRACE(logger_);
   LOG4CXX_DEBUG(logger_, "Timer is to be destroyed " << name_);
-  Stop();
   sync_primitives::AutoLock auto_lock(task_lock_);
   DCHECK(task_);
   delete task_;
+  if (thread_) {
+    thread_->join();
+    DCHECK(delegate_);
+    delete delegate_;
+    threads::DeleteThread(thread_);
```

Comment:
For what? It's destructor. After it object doesn't exist and his memory will use other processes.